src/dotnet/commands/dotnet-test/DotnetTest.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Testing.Abstractions;
+
+namespace Microsoft.DotNet.Tools.Test
+{
+    public class DotnetTest : IDotnetTest
+    {
+        private readonly IList<IReportingChannel> _channels;
+        private readonly IList<IDotnetTestMessageHandler> _messageHandlers;
+        private readonly ITestMessagesCollection _messages;
+
+        public IDotnetTestMessageHandler TestSessionTerminateMessageHandler { private get; set; }
+        public IDotnetTestMessageHandler UnknownMessageHandler { private get; set; }
+
+        public DotnetTestState State { get; private set; }
+
+        public string AssemblyUnderTest { get; }
```

Comment:
not sure about the name here. It implied that this would be an Assembly, not a String.